taichi/analysis/clone.cpp

```diff
@@ -134,6 +134,9 @@ class IRCloner : public IRVisitor {
 
     if (kernel != nullptr) {
       new_root->kernel = kernel;
+    } else if (new_root->get_parent() == nullptr &&
+               new_root->get_kernel() == nullptr) {
+      new_root->kernel = root->get_kernel();
```

Comment:
I think with these lines, we won't need to pass the `Kernel *` argument. The reason why we passed it is the `OffloadedStmt` is usually under a block initially (so it doesn't store the `Kernel *`) but will become an IR root later.